libraries/AP_LTM_Telem/AP_LTM_Telem.cpp

```diff
@@ -0,0 +1,230 @@
+ *   A Frame (Attitude) (5 Hz): 10BYTES
+ *     0x24 0x54 0x41 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xC0
+ *      $     T   A   --PITCH-- --ROLL--- -HEADING-  CRC
+ *   S Frame (Sensors) (2 Hz): 11BYTES
+ *     0x24 0x54 0x53 0xFF 0xFF  0xFF 0xFF    0xFF    0xFF      0xFF       0xC0
+ *      $     T   S   VBAT(mV)  Current(mA)   RSSI  AIRSPEED  ARM/FS/FMOD   CRC
+ * ################################################################################################################# */
+
+#include "AP_LTM_Telem.h"
+
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_BattMonitor/AP_BattMonitor.h>
+#include <AP_Notify/AP_Notify.h>
+#include <AP_RSSI/AP_RSSI.h>
+
+extern const AP_HAL::HAL& hal;
+
+// init - perform required initialisation
+void AP_LTM_Telem::init()
+{
+    const AP_SerialManager &serial_manager = AP::serialmanager();
+    
+    // check for LTM_Port
+    if ((_port = serial_manager.find_serial(
+            AP_SerialManager::SerialProtocol_LTM_Telem, 0))) {
+        _port->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
+        // initialise UART
+        _port->begin(AP_SERIALMANAGER_LTM_BAUD, AP_SERIALMANAGER_LTM_BUFSIZE_RX,
+        AP_SERIALMANAGER_LTM_BUFSIZE_TX);
+        hal.scheduler->register_io_process(
+                FUNCTOR_BIND_MEMBER(&AP_LTM_Telem::tick, void));
+    }
+}
+
+void AP_LTM_Telem::send_LTM(uint8_t lt_packet[], uint8_t lt_packet_size)
+{
+    // calculate checksum
+    uint8_t lt_crc = 0x00;
+    for (uint8_t i = 3; i < lt_packet_size - 1; i++) {
+        lt_crc ^= lt_packet[i];
+    }
+    lt_packet[lt_packet_size - 1] = lt_crc;
+    _port->write(lt_packet, lt_packet_size);
+}
+
+// GPS frame
+void AP_LTM_Telem::generate_Gframe(void)
+{
+    int32_t lat;                        // latitude
+    int32_t lon;                        // longtitude
+    uint8_t gndspeed;                   // gps ground speed
+    int32_t alt;                        // altitude
+    uint8_t sats_visible;               // number of visible gps sats
+    uint8_t fix_type;                   // gps fix type
+    Location loc;
+    float alt_ahrs;
+    const AP_GPS &gps = AP::gps();
+    const AP_AHRS &ahrs = AP::ahrs();
+
+    ahrs.get_relative_position_D_home(alt_ahrs);
+
+    if (ahrs.get_position(loc)) {
+        lat = loc.lat;
+        lon = loc.lng;
+        gndspeed = (uint8_t) roundf(gps.ground_speed());
+        alt = (int32_t) roundf(alt_ahrs);
+        sats_visible = (uint8_t) AP::gps().num_sats();
+        fix_type = (uint8_t) AP::gps().status();
+    } else {
+        lat = 0;
+        lon = 0;
+        gndspeed = 0;
+        alt = (int32_t) roundf(alt_ahrs);
+        sats_visible = (uint8_t) AP::gps().num_sats();
+        fix_type = (uint8_t) AP::gps().status();
+    }
+
+    uint8_t lt_buff[LTM_GFRAME_SIZE];
+    // protocol: START(2 bytes)FRAMEID(1byte)LAT(cm,4 bytes)LON(cm,4bytes)SPEED(m/s,1bytes)ALT(cm,4bytes)SATS(6bits)FIX(2bits)CRC(xor,1byte)
```

Comment:
```suggestion
        static_assert(LTM_GFRAME_SIZE >= 18, "GRAME buffer too small");
        // protocol: START(2 bytes)FRAMEID(1byte)LAT(cm,4 bytes)LON(cm,4bytes)SPEED(m/s,1bytes)ALT(cm,4bytes)SATS(6bits)FIX(2bits)CRC(xor,1byte)
```

Honestly, this is probably not worth it.

Reply:
Ok, I'm ignoring it ;)